### ISSUE TITLE
Simple set-up

### ISSUE THREAD

**Issue by RobDolinMS** [opencontainers/validator.opencontainers.org#5]
A very simple set-up for this might be:
- [ ] Provision the "validator" subdomain for opencontainers.org
- [ ] Set-up validator.opencontainers.org to point to a URL like http://opencontainers.github.io/image-tools/ 
- [ ] Publish code into gh-pages branch of OCI images-tools repo to support a web interface to OCI Image Format validation (this is non-trivial)

**Comment by vbatts** [comment]
Yeah, the first step here will be a largely trivial validation of a pasted json blob. But the further validation for validating an image (might including a user uploading arbitrary content) or providing a runtime bundle that will validate a runtime (user downloading content), these require trust and moving parts.

I'm almost of the mind to have the site generate a session token at user request, and provide a cli that submits the results set using that token, which they can then share the results.

**Comment by stevvooe** [comment]
@vbatts Do mean in the pastebin/play.golang.org sense?

**Comment by wking** [comment]
On Fri, Aug 26, 2016 at 11:00:29AM -0700, Vincent Batts wrote:
> Yeah, the first step here will be a largely trivial validation of a
> pasted json blob.

This should be somewhat straightfoward.  There are lots of JavaScript
JSON Schema validators, but we'd have to write additional JavaScript
to handle the requirements we can't represent in JSON Schema.  That
duplicates some of the effort that's already gone into encoding those
requirements in Go [1], and host-specific checks may be difficult to
do from the browser, but it's probably worth the effort to avoid
needing moving parts on the server side (where you couldn't do
host-specific checks anyway).

> I'm almost of the mind to have the site generate a session token at
> user request, and provide a cli that submits the results set using
> that token, which they can then share the results.

I think an implementation that uses an OCI-managed Buildbot pool (or
some such) to handle validation would work.  But I'm not clear on how
you'd convince visitors that results are reliable if the results had
been submitted by non-OCI users.

If the idea is to replace the Buildbot component with manual testing
(so OCI testers pick up the submission and token, run the validation
locally, and submit the results), that would help with perceived
reliability (vs. the submitter doing their own validation and
uploading the results).  But it sounds like a tedious job, and latency
would probably be pretty bad.

And regardless of whether it's Buildbot or humans running the test
suite, you're still going to need a database or some such to store the
queue of submitted jobs and the results of completed jobs.

[1]: https://github.com/opencontainers/ocitools/blob/master/cmd/ocitools/validate.go
